dracut/30ignition/ignition-diskful-udev.service

```diff
@@ -0,0 +1,25 @@
+[Unit]
+Description=Ignition Diskful UDEV Trigger
+Documentation=https://github.com/coreos/ignition
+
+Before=ignition-diskful.target
+Before=ignition-diskful-subsequent.target
+Before=ignition-subsequent.target
+
+After=ignition-setup-user.service
+After=ignition-setup-base.service
+
+# Wait for multipathd
+Wants=multipathd.service
+After=multipathd.service
+
+# We need sysinit.target to be ready in order to use udev
+# triggers. Otherwise triggering will have no effect.
+Requires=sysinit.target systemd-udevd.service
+After=sysinit.target systemd-udevd.service systemd-udev-settle.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+# Have to shell out since we need shell expansion of the *
+ExecStart=-/bin/sh -c '/usr/bin/udevadm trigger -w -v /dev/dm*'
```

Comment:
I think what I find confusing here is I'd really expect there to be an explicit command for this somewhere else.  Is there not similar code in dracut or multipathd?

Reply:
Yeah, that's my confusion as well. I had hoped that we'd get this for free and all we really had to do was just make sure that the units we have which need ro access to `/boot` run early enough that they can't interfere with multipath rewiring the symlinks. Poking around on this to see how it happens on traditional.

Reply:
The multipath systemd unit triggers `systemd-udev-trigger`. The problem we have is we need to wait until the symlinks are updated (which won't happen if the the disk is mounted) and during testing we had a racing condition. That's why I have this ugly workaround.